### ISSUE TITLE
Better long term solution for button fix needed

### ISSUE THREAD

**Issue by ProgrammerDan** [DevotedMC/ExilePearl#11]
https://github.com/DevotedMC/ExilePearl/blob/master/src/main/java/com/devotedmc/ExilePearl/listener/PlayerListener.java#L834

See notes.

**Comment by TealNerd** [comment]
Is this still an issue?

**Comment by ProgrammerDan** [comment]
afaik yes, haven't tested in a while and don't remember the current status.

**Comment by TealNerd** [comment]
What was wrong?

**Comment by ProgrammerDan** [comment]
Comments are like 5-6 lines below the one I highlight, but basically, you couldn't craft buttons. I think I fixed that sort of? But felt that my solution wasn't great. Been over a year now though...

**Comment by TealNerd** [comment]
Yeah I think a return there is fine, maybe squashing those couple cases into one if statement would be good though

**Comment by ProgrammerDan** [comment]
Wouldn't really change whatever past-me had an issue with though :3

**Comment by TealNerd** [comment]
Good point

**Comment by danquist** [comment]
I think previously it used to cancel the craft event if it was a button, but now it just returns, or something along those lines. I don't exactly remember, but there was some initial attempt made to fix the bug.

The issue was never closed because it's still kind of hacky, I couldn't ever think of a better solution.
